CalibCalorimetry/EBPhase2TPGTools/plugins/EcalEBPhase2TPParamProducer.cc

```diff
@@ -0,0 +1,350 @@
+#include "Geometry/CaloGeometry/interface/CaloCellGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "Geometry/EcalMapping/interface/EcalElectronicsMapping.h"
+#include "Geometry/EcalMapping/interface/EcalMappingRcd.h"
+#include "DataFormats/EcalDigi/interface/EcalConstants.h"
+#include "DataFormats/EcalDetId/interface/EcalSubdetector.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+
+#include "CalibCalorimetry/EBPhase2TPGTools/plugins/EcalEBPhase2TPParamProducer.h"
+
+#include <iostream>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <ctime>
+#include <fstream>
+#include <iomanip>
+
+const int EcalEBPhase2TPParamProducer::linTopRange_ = 16383;
+
+EcalEBPhase2TPParamProducer::EcalEBPhase2TPParamProducer(edm::ParameterSet const& pSet)
+    : theBarrelGeometryToken_(esConsumes(edm::ESInputTag("", "EcalBarrel"))),
+      inFile_(pSet.getUntrackedParameter<std::string>("inputFile")),
+      outFile_(pSet.getUntrackedParameter<std::string>("outputFile")),
+      nSamplesToUse_(pSet.getParameter<unsigned int>("nSamplesToUse")),
+      useBXPlusOne_(pSet.getParameter<bool>("useBXPlusOne")),
+      phaseShift_(pSet.getParameter<double>("phaseShift")),
+      nWeightGroups_(pSet.getParameter<unsigned int>("nWeightGroups")),
+      theEcalTPGPedestals_Token_(esConsumes(edm::ESInputTag("EcalLiteDTUPedestals", ""))),
+      et_sat_(pSet.getParameter<double>("Et_sat")),
+      xtal_LSB_(pSet.getParameter<double>("xtal_LSB")),
+      binOfMaximum_(pSet.getParameter<unsigned int>("binOfMaximum"))
+
+{
+  out_file_ = gzopen(outFile_.c_str(), "wb");
+
+  TFile* inFile = new TFile(inFile_.c_str(), "READ");
+
+  inFile->GetObject("average-pulse", thePulse_);
+  delete inFile;
+}
+
+EcalEBPhase2TPParamProducer::~EcalEBPhase2TPParamProducer() { gzclose(out_file_); }
+
+void EcalEBPhase2TPParamProducer::beginJob() {}
+
+void EcalEBPhase2TPParamProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.addUntracked<std::string>("inputFile");
+  desc.addUntracked<std::string>("outputFile");
+  desc.add<unsigned int>("nSamplesToUse", 8);
+  desc.add<bool>("useBXPlusOne", false);
+  desc.add<double>("phaseShift", 2.581);
+  desc.add<unsigned int>("nWeightGroups", 61200);
+  desc.add<double>("Et_sat", 1998.36);
+  desc.add<double>("xtal_LSB", 0.0488);
+  desc.add<unsigned int>("binOfMaximum", 6);
```

Comment:
Same comment than in EcalEBTrigPrimPhase2Producer.cc. Assuming that this needs to be the same than there is would be better to use a centrally set constant.